hazelcast/src/main/java/com/hazelcast/internal/partition/operation/UrgentPartitionRunnable.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.partition.operation;
+
+import com.hazelcast.spi.impl.InternalCompletableFuture;
+import com.hazelcast.spi.impl.PartitionSpecificRunnable;
+import com.hazelcast.spi.impl.operationservice.UrgentSystemOperation;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+
+final class UrgentPartitionRunnable<T>
+        implements PartitionSpecificRunnable, UrgentSystemOperation {
+
+    final CountDownLatch done = new CountDownLatch(1);
```

Comment:
there is no any awaiting on this latch, no?

Reply:
Await happens in client code that needs to wait for the runnable/callable to complete eg [here](https://github.com/hazelcast/hazelcast/pull/19013/files/a55aa84666f114372592a2f1bf2fb53ed0633932..ccc2cb8b9aa3d32d3bbb555eb3ca7180eb345687#diff-2ccd9b76b1a1019e2129104193e9b88e7cb7b239062a80587b429231b3c520c2R121).

But now I remember that I also implemented return of `Callable`'s result via the `UrgentPartitionRunnable#future`, so there is no need to have both ways to sync on the future's completion; I will push a commit to remove the latch and just `join` on the future instead of `await`.

Reply:
Ohh, got it. thanks :)